Changes for systemd (#73)
owncloud/docs-ocis: modules/ROOT/pages/deployment/manual/manual-setup.adoc

kill -15 221706
----

=== Setting up systemd for oCIS

To run the oCIS runtime as a {systemd-url}[systemd] service, create the file `/etc/systemd/system/ocis.service` with the following content. The easiest way to do this is:

[source,bash]
----
sudo systemctl edit ocis.service
----

Then copy the content of the systemd file below into the editor and save it.

* The user `ocisadmin` is just a placeholder for any user to run oCIS, the user must exist or be created first.
+
NOTE: This user can be a system user without e.g. having a home directory.
* We strongly advise *against* using the user `root` for this purpose.
* Placing the environment file in `/etc/ocis/` is only a suggestion, but a good one.

> ** The directory `/etc/ocis/` must exist and the user must be able to read. For security reasons, this user should have restricted permissions. In case, create and set permissions, adopt the values to your needs:

Reviewer: ```suggestion
** The directory `/etc/ocis/` must exist and the `ocisadmin` must be able to read it. For security reasons, this user should have restricted permissions. Create and set permissions, adjust the values to your needs:
```

WantedBy=multi-user.target
----

> Now create the file `/etc/ocis/ocis.env` with the definitions of environment variables. See following sections for additional environment variables like xref:deployment/general/general-info.adoc#accessing-ocis-via-the-webui[Accessing oCIS Via the WebUI] or xref:deployment/general/general-info.adoc#define-the-ocis-data-path[Define the oCIS Data Path].

Reviewer: ```suggestion
Now create the file `/etc/ocis/ocis.env` with the definitions of environment variables. See the following sections for additional environment variables like xref:deployment/general/general-info.adoc#accessing-ocis-via-the-webui[Accessing oCIS Via the WebUI] or xref:deployment/general/general-info.adoc#define-the-ocis-data-path[Define the oCIS Data Path].
```

NOTE: This is just an example with a minimal set of environment variables used.

[source,plaintext]
----
OCIS_INSECURE=true
OCIS_URL=https://localhost:9200
PROXY_HTTP_ADDR=0.0.0.0:9200

OCIS_LOG_LEVEL=error
----

Run the following command to apply your changes:

[source,bash]
----
sudo systemctl daemon-reload
----

Now you can run oCIS as a systemd service. Start it with:

[source,bash]
----
sudo systemctl enable --now ocis
----

> With this setup, oCIS is restarted automatically if the host got rebooted.

Reviewer: ```suggestion
With this setup, oCIS is restarted automatically after a reboot.
```
